tests/ArraySchemaValidatorTest.php

```diff
@@ -36,28 +36,26 @@ public function testArrayOfStringsValid(): void {
   }
 
   public function testArrayOfStringsLegacyArrays(): void {
-    $validator = new ArraySchemaValidator(darray['array_of_strings' => varray['test', 'list', 'of', 'strings']]);
+    $validator = new ArraySchemaValidator(darray['array_of_strings' => vec['test', 'list', 'of', 'strings']]);
     $validator->validate();
 
     expect($validator->isValid())->toBeTrue();
     $validated = $validator->getValidatedInput();
     expect(C\count($validated['array_of_strings'] ?? vec[]))->toBeSame(4);
   }
 
-  public function testArrayOfStringsLegacyAndHackArrays(): void {
+  public function testDictOfStringsInvalid(): void {
     $validator = new ArraySchemaValidator(dict[
-      'array_of_strings' => varray['test', 'list', 'of', 'strings'],
+      'array_of_strings' => dict['test' => 'test'],
     ]);
     $validator->validate();
 
-    expect($validator->isValid())->toBeTrue();
-    $validated = $validator->getValidatedInput();
-    expect(C\count($validated['array_of_strings'] ?? vec[]))->toBeSame(4);
+    expect($validator->isValid())->toBeFalse();
   }
 
   public function testUntypedArrayValid(): void {
     $validator = new ArraySchemaValidator(dict[
-      'untyped_array' => varray['test', 'values'],
+      'untyped_array' => vec['test', 'values'],
```

Comment:
What happens with varray? Does it throw an error?

Reply:
yeah it does